File: 0242-valid-anagram/solution.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.length() != t.length())return false;

        int charCount[256] = {0};

        for (int i = 0; i<s.length(); i++){
            charCount[s[i]]++;
            charCount[t[i]]--;
        }

        for (auto count : charCount){
            if(count != 0)return false;
        }
        return true;
    }
};

File: 0242-valid-anagram/solution.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if(s.length() != t.length())return false;
        int size = s.length();
        int[] arr = new int[26];
        for (int i = 0; i<size; i++){
            arr[s.charAt(i) - 'a']++;
            arr[t.charAt(i) - 'a']--;
        }
        for (int i = 0; i<26; i++)if(arr[i] != 0)return false;
        return true;
    }
}
